### PR TITLE
Match 3 effects

### PR DESCRIPTION
**Pull request by savaughn**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.02%) on `us`.

**Review by nanaian — APPROVED** [review]

File: src/effects/hieroglyphs.c [hunk 1]
```diff
@@ -8,7 +8,31 @@ INCLUDE_ASM(s32, "effects/hieroglyphs", hieroglyphs_main);
...
+    if (effect->flags & 16) {
+        effect->flags &= ~16;
+        data->unk_10 = 16;
+    }
+
+    if (data->unk_10 < 1000) {
+        data->unk_10--;
+    }
+    temp_a2 = data->unk_14 + 1; // Needed for match?
+    data->unk_14++;
```
Comment (Brotenko) [inline src/effects/hieroglyphs.c:24]:
```suggestion
    temp_a2 = ++data->unk_14;
```

File: src/effects/huff_puff_breath.c [hunk 2]
```diff
@@ -8,7 +8,46 @@ INCLUDE_ASM(s32, "effects/huff_puff_breath", huff_puff_breath_main);
...
+    if (temp_a1 < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+    if (temp_a1 < 16) {
+        data->unk_24 = temp_a1 * 16;
+    }
+    if (temp_a2 < 16) {
+        data->unk_24 = (temp_a2 * 16) + 15;
+    }
```
Comment (Brotenko) [inline src/effects/huff_puff_breath.c:35]:
```suggestion
    temp_a2 = ++data->unk_14;

    if (temp_a1 < 0) {
        shim_remove_effect(effect);
        return;
    }
    if (temp_a1 < 16) {
        data->unk_24 = data->unk_10 * 16;
    }
    if (temp_a2 < 16) {
        data->unk_24 = (data->unk_10 * 16) + 15;
    }
```

**Review by Brotenko — CHANGES_REQUESTED** [review]

File: src/effects/huff_puff_breath.c (hunk 2 above)
Comment (Brotenko) [inline src/effects/huff_puff_breath.c:35]:
```suggestion
    temp_a2 = ++data->unk_14;
    
    if (data->unk_10 < 0) {
        shim_remove_effect(effect);
        return;
    }
    if (data->unk_10 < 16) {
        data->unk_24 = data->unk_10 * 16;
    }
    if (temp_a2 < 16) {
        data->unk_24 = (temp_a2 * 16) + 15;
    }
```

[See here](https://decomp.me/scratch/hezDG)

Comment (ethteck) [reply src/effects/huff_puff_breath.c:35]: I'd prefer postincrementing the field and then storing to a temp. I'm gonna play with this locally and see if I can clean it up (also applies above)

File: src/effects/hieroglyphs.c (hunk 1 above)
Comment (ethteck) [reply src/effects/hieroglyphs.c:24]: I agree

File: src/effects/hieroglyphs.c [hunk 3]
```diff
@@ -8,7 +8,31 @@ INCLUDE_ASM(s32, "effects/hieroglyphs", hieroglyphs_main);
...
+    HieroglyphsFXData* data = effect->data.hieroglyphs;
+    s32 temp_a2;
+
+    if (effect->flags & 16) {
+        effect->flags &= ~16;
+        data->unk_10 = 16;
+    }
+
+    if (data->unk_10 < 1000) {
+        data->unk_10--;
```
Comment (ethteck) [inline src/effects/hieroglyphs.c:21]: we call these temps timeLeft (and the ones that count up every tick lifeTime)

File: src/effects/sweat.c [hunk 4]
```diff
@@ -8,7 +8,24 @@ INCLUDE_ASM(s32, "effects/sweat", sweat_main);
 void sweat_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/sweat", sweat_update);
+void sweat_update(EffectInstance *effect) {
+    SweatFXData* data = effect->data.sweat;
+
+    data->unk_24--;
```
Comment (ethteck) [inline src/effects/sweat.c:14]: timeLeft

File: src/effects/hieroglyphs.c (hunk 1 above)
Comment (ethteck) [reply src/effects/hieroglyphs.c:24]: and unk_14 can be lifeTime

File: src/effects/huff_puff_breath.c (hunk 2 above)
Comment (ethteck) [reply src/effects/huff_puff_breath.c:35]: additionally, unk14 is lifeTime and unk_10 is timeLeft

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/effects/huff_puff_breath.c (hunk 2 above)
Comment (Brotenko) [reply src/effects/huff_puff_breath.c:35]: Oh, and you can remove the declaration for `temp_a1` on line 13

File: src/effects/hieroglyphs.c (hunk 1 above)
Comment (savaughn) [reply src/effects/hieroglyphs.c]: done

File: src/effects/hieroglyphs.c (hunk 3 above)
Comment (savaughn) [reply src/effects/hieroglyphs.c]: done

File: src/effects/sweat.c (hunk 4 above)
Comment (savaughn) [reply src/effects/sweat.c]: done

**Review by Brotenko — APPROVED** [review]
Luuks grud

**Pull request merged**